src/systems/systems/src/hydraulic/electrical_generator.rs

```diff
@@ -0,0 +1,670 @@
+        self.update_speed(context);
+        self.update_flow(context);
+    }
+
+    fn update_valve_position(
+        &mut self,
+        context: &UpdateContext,
+        gcu_interface: &impl ControlValveCommand,
+    ) {
+        self.valve
+            .update(context, gcu_interface.valve_position_command());
+    }
+
+    fn update_resistant_torque(&mut self, emergency_generator: &impl EmergencyGeneratorInterface) {
+        let resistant_torque = if self.speed().get::<radian_per_second>() < 1.
+            || self.virtual_displacement < Volume::new::<cubic_inch>(0.001)
+        {
+            Torque::new::<newton_meter>(Self::STATIC_RESISTANT_TORQUE_WHEN_UNPOWERED_NM)
+        } else {
+            let theoretical_torque = Torque::new::<newton_meter>(
+                emergency_generator.generated_power().get::<watt>()
+                    / self.speed().get::<radian_per_second>(),
+            );
+            theoretical_torque + (1. - Self::EFFICIENCY) * theoretical_torque
+        };
+
+        self.total_torque -= resistant_torque;
+    }
+
+    fn update_generated_torque(&mut self, pressure: Pressure) {
+        self.virtual_displacement = self.virtual_displacement_after_valve_inlet();
+
+        self.generated_torque = Torque::new::<pound_force_inch>(
+            pressure.get::<psi>() * self.virtual_displacement.get::<cubic_inch>()
+                / (2. * std::f64::consts::PI),
+        );
+    }
+
+    fn virtual_displacement_after_valve_inlet(&mut self) -> Volume {
+        self.displacement * self.valve.position()
+    }
+
+    fn update_speed(&mut self, context: &UpdateContext) {
+        let friction_torque = Torque::new::<newton_meter>(
+            Self::DYNAMIC_FRICTION_TORQUE_CONSTANT * -self.speed.get::<revolution_per_minute>(),
+        );
+
+        self.total_torque += friction_torque;
+        self.total_torque += self.generated_torque;
+
+        self.acceleration = AngularAcceleration::new::<radian_per_second_squared>(
+            self.total_torque.get::<newton_meter>() / self.inertia,
+        );
+        self.speed += AngularVelocity::new::<radian_per_second>(
+            self.acceleration.get::<radian_per_second_squared>() * context.delta_as_secs_f64(),
+        );
+        self.speed = self
+            .speed
+            .max(AngularVelocity::new::<revolution_per_minute>(0.));
+
+        self.total_torque = Torque::new::<newton_meter>(0.);
+    }
+
+    fn update_flow(&mut self, context: &UpdateContext) {
+        self.current_flow = self.flow();
+
+        let total_volume = self.current_flow * context.delta_as_time();
+        self.total_volume_to_actuator += total_volume;
+        self.total_volume_to_reservoir = self.total_volume_to_actuator;
+    }
+
+    fn flow(&self) -> VolumeRate {
+        VolumeRate::new::<gallon_per_minute>(
+            self.speed.get::<revolution_per_minute>()
+                * self.virtual_displacement.get::<cubic_inch>()
+                / 231.,
```

Comment:
**thought**: I remember seeing this number before. However, it is also a bit magic... Could you maybe make it a local const or something?